src/receipt.rs

```diff
@@ -53,6 +53,7 @@ pub enum Receipt {
         reason: InstructionResult,
         pc: Word,
         is: Word,
+        contract_id: Option<ContractId>,
```

Comment:
Issue says:
> As noted in the specs, this field should not be included in canonical serialization. It's only an application-level concern and shouldn't be validated as part of the blockchain state. (ie. exclude from derive(eq) etc)

`Receipt` derives `PartialEq`, `Eq`, and `Hash`. And it seems we don't need to take into account optional fields.

Also, maybe it is related to serde.

But I don't know because `Option<Metadata>` in the `Transaction` has the same problem=) 
<img width="709" alt="image" src="https://user-images.githubusercontent.com/18346821/193608271-b6f624a8-e009-46d8-b17e-cef8c2592517.png">


@Voxelot @vlopes11 Do we need to do the manual implementation for the `Receipt` and `Transaction` of all derived traits?

Reply:
we can just provide `#[serde(skip)]` to the fields that we don't want to include in (de)serialization

Reply:
You could use [Derivative](https://github.com/mcarton/rust-derivative) to do `#[derivative(Eq="ignore")]` etc. It does add a dependency but it's a pretty minimal proc_macro dep and it avoids issue where manually impl `PartialEq` doesn't match the manual impl `Hash` etc.

Reply:
@Salka1988 Could you also do the same for `Transaction.metadata` please?

Reply:
Added!